starport/interface/cli/starport/cmd/network_chain_join.go

```diff
@@ -28,13 +39,21 @@ func NewNetworkChainJoin() *cobra.Command {
 		Args:  cobra.ExactArgs(1),
 	}
 	c.Flags().AddFlagSet(flagSetHomes())
+	c.Flags().String(flagGentx, "", fmt.Sprintf("Path to a gentx file (optional, requires --%s to be provided)", flagGentx))
+	c.Flags().String(flagPeer, "", fmt.Sprintf("Configure peer in node-id@host:port format (optional, requires --%s to be provided)", flagPeer))
 	return c
 }
 
 func networkChainJoinHandler(cmd *cobra.Command, args []string) error {
-	var (
-		chainID = args[0]
-	)
+	chainID := args[0]
+
+	manualFlagCheck, err := xcobra.CheckFlagValues(cmd.Flags(), flagGroupManualJoin)
+	if err != nil {
+		return err
+	}
+	if manualFlagCheck == xcobra.FlagsSomeEmpty {
```

Comment:
Mmh is there a practical reason we must prevent using only one flag?

Reply:
this is what's proposed in the issue https://github.com/tendermint/starport/issues/882#issue-833638182, can't remember much what led us to have this decision. cc @fadeev 

I agree, seems like it's more beneficial to not make them mutually exclusive.

Reply:
If a user uses `--gentx` it means we're not doing the joining process automatically, so they need to provide `--peer`.

Reply:
Technically, it should be a separate command because `gentx` and `peer` are required arguments, but from the user's perspective it's still "joining the chain".

Reply:
Denis, can you please elobarate the part related to `means we're not doing the joining process automatically, so they need to provide --peer`?

In-fact, we may not need to introduce a `--peer` flag at all in order to manual join work since we're already prompting for it interactively.

Reply:
@ilgooz ok, you're right. I think the logic can be: we prompt for it, unless `--peer` flag is used (in which case we use the value from the flag).

Reply:
solved by 941f52fc3071a645f3009d3e5c768b6f0bd28603.